### PR TITLE
[stable/prometheus-operator] Add release namespace to header in sync scripts

### PR DESCRIPTION
**Pull request by mercul3s**
#### What this PR does / why we need it:
This PR adds the release namespace to grafana dashboards and alerting rules generated from `sync_grafana_dashboards.py` and `sync_prometheus_rules.py`. This ensures that the rules and dashboards configmaps are deployed to the same namespace as prometheus and grafana.

Note: This change does not contain updates to the templates, as I was unsure if that was necessary at this point and ran into some issues with updated alertmanager rules which I don't have bandwidth to debug at this time (I've disabled them in our deployment for now). I can resubmit with updated templates if desired.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/prometheus-operator/hack/sync_grafana_dashboards.py [hunk 1]
```diff
@@ -77,6 +77,7 @@ def new_representer(dumper, data):
 kind: ConfigMap
 metadata:
   name: {{ printf "%%s-%%s" (include "prometheus-operator.fullname" $) "%(name)s" | trunc 63 | trimSuffix "-" }}
+  namespace: {{ .Release.Namespace }}
```
Comment (bismarck) [inline stable/prometheus-operator/hack/sync_grafana_dashboards.py]: Could you make this `$.Release.Namespace`?

File: stable/prometheus-operator/hack/sync_prometheus_rules.py [hunk 2]
```diff
@@ -117,6 +117,7 @@ def new_representer(dumper, data):
 kind: PrometheusRule
 metadata:
   name: {{ printf "%%s-%%s" (include "prometheus-operator.fullname" .) "%(name)s" | trunc 63 | trimSuffix "-" }}
+  namespace: {{ .Release.Namespace }}
```
Comment (bismarck) [inline stable/prometheus-operator/hack/sync_prometheus_rules.py]: Could you make this `$.Release.Namespace`?

**Review by bismarck — CHANGES_REQUESTED** [review]
@mercul3s  Thanks for the PR!  Could you try regenerating the rules and dashboards with the updated scripts and include those in your PR?

**Comment by bismarck** [comment]
@mercul3s Looks like you're going to have to bump the chart version one more time.  Everything else looks good.

**Comment by mercul3s** [comment]
@bismarck thanks for taking a look! The chart version is up to date, it looks like the linter is failing on one of the charts generated from the upstream:
```
Linting chart with values file 'stable/prometheus-operator/ci/01-provision-crds-values.yaml'...

==> Linting stable/prometheus-operator
[ERROR] templates/: parse error in "prometheus-operator/templates/prometheus/rules-1.14/prometheus.yaml": template: prometheus-operator/templates/prometheus/rules-1.14/prometheus.yaml:171: bad character U+007B '{'

Error: 1 chart(s) linted, 1 chart(s) failed
```
I'm taking a look at it, but it's not obvious to me why it's failing the lint step there as it looks like it should be valid.

**Comment by bismarck** [comment]
@mercul3s Looks like the script isn't encoding the prometheus query in the description properly (escaping a template in a template makes my head explode).

@vsliouniaev Do you think someone manually fixed the encoding [here](https://github.com/helm/charts/pull/19137/files#diff-365cff774eb3b04dae4571b2b53cbcb2L171)?

**Comment by mercul3s** [comment]
@bismarck I was just going down that path and yes, it makes my head explode too. I'm guessing it's getting mangled in this step: https://github.com/helm/charts/blob/master/stable/prometheus-operator/hack/sync_prometheus_rules.py#L92-L119

**Pull request closed without merging**